### PR TITLE
service: Add eos-companion-app.service and socket activation

### PR DESCRIPTION
**Pull request by smspillaz**
This will listen on port 1110 to launch the Companion App, if available.

https://phabricator.endlessm.com/T20130

### CODE REVIEW

**Comment by smspillaz** [comment]
Something I'm not sure about - should we have a wrapper executable that returns 0 in case the flatpak isn't installed?

**Comment by cosimoc** [comment]
If the app is not installed, I'd expect the systemd unit to error out, so I don't think we need that

File: tmpfiles.d/eos-companion-app-helper.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Type Path    Mode UID  GID  Age Argument
+d /var/lib/companion-app-helper 0755 companion-app-helper companion-app-helper -
+Z /var/lib/companion-app-helper 0755 companion-app-helper companion-app-helper -
```
Comment (cosimoc) [inline tmpfiles.d/eos-companion-app-helper.conf]: How is this file installed? Does not seem to get installed from anywhere.

Comment (smspillaz) [reply tmpfiles.d/eos-companion-app-helper.conf]: Oops, I must have forgotten to add my changes to `Makefile.am`

Comment (smspillaz) [reply tmpfiles.d/eos-companion-app-helper.conf]: Okay, fixed 😅

File: Makefile.am [hunk 2]
```diff
@@ -79,6 +81,11 @@ dist_polkitrules_DATA = \
 	10-allow-mounting-endless-image-device.rules \
 	$(NULL)
 
+tmpfilesddir = $(libdir)/tmpfiles.d
```
Comment (cosimoc) [inline Makefile.am]: We can't use `$(libdir)` here unfortunately. See e.g. this commit: https://github.com/endlessm/eos-metrics-instrumentation/commit/d4953db3bab8ffa55694849acf884c704ae144bb#diff-8d777f385d3dfec8815d20f7496026dc

Comment (smspillaz) [reply Makefile.am]: Ah, frustrating. Thanks for catching!

**Pull request merged**